packages/region/src/services/region-module.ts

```diff
@@ -143,54 +143,72 @@ export default class RegionModuleService<
     return await this.regionService_.create(normalizedDbRegions, sharedContext)
   }
 
-  async update(
-    selector: FilterableRegionProps,
-    data: UpdatableRegionFields,
+  async upsert(
+    data: UpsertRegionDTO[],
     sharedContext?: Context
   ): Promise<RegionDTO[]>
-  async update(
-    regionId: string,
-    data: UpdatableRegionFields,
+  async upsert(
+    data: UpsertRegionDTO,
     sharedContext?: Context
   ): Promise<RegionDTO>
-  async update(data: UpdateRegionDTO[]): Promise<RegionDTO[]>
   @InjectManager("baseRepository_")
-  async update(
-    idOrSelectorOrData: string | FilterableRegionProps | UpdateRegionDTO[],
-    data?: UpdatableRegionFields,
+  async upsert(
+    data: UpsertRegionDTO | UpsertRegionDTO[],
     @MedusaContext() sharedContext: Context = {}
   ): Promise<RegionDTO | RegionDTO[]> {
-    const updateResult = await this.update_(
-      idOrSelectorOrData,
-      data,
-      sharedContext
-    )
-
-    const regions = await this.baseRepository_.serialize<
-      RegionDTO[] | RegionDTO
-    >(updateResult)
-
-    return isString(idOrSelectorOrData) ? regions[0] : regions
-  }
+    const input = Array.isArray(data) ? data : [data]
+    const forUpdate = input.filter((region) => region.id)
```

Comment:
@adrien2p this doesn't quite work since the `UpsertRegionDTO` object has several undefined values that are required in `UpdateRegionDTO` and `CreateRegionDTO`, so we have to force the type.

 I guess what I can do is also apply some validation on the input since of course you can never trust just the typings, but I thought that's better reserved for the create and update functions themselves.
 
 I am not a Typescript expert so there might be another better way to do it?

Reply:
sorry I wanted to put `UpdateRegionInput` instead, of course we can also have some validation but in theory the subsequent method will handle it